lib/resourcemerge/operators_test.go

```diff
@@ -0,0 +1,235 @@
+				},
+			},
+
+			expectedModified: true,
+			expected: operatorsv1.OperatorGroup{
+				Spec: operatorsv1.OperatorGroupSpec{
+					ServiceAccountName: "bar",
+				},
+			},
+		},
+		{
+			name: "same static provided apis",
+			existing: operatorsv1.OperatorGroup{
+				Spec: operatorsv1.OperatorGroupSpec{
+					StaticProvidedAPIs: true,
+				},
+			},
+			required: operatorsv1.OperatorGroup{
+				Spec: operatorsv1.OperatorGroupSpec{
+					StaticProvidedAPIs: true,
+				},
+			},
+
+			expectedModified: false,
+			expected: operatorsv1.OperatorGroup{
+				Spec: operatorsv1.OperatorGroupSpec{
+					StaticProvidedAPIs: true,
+				},
+			},
+		},
+		{
+			name: "different static provided apis",
+			existing: operatorsv1.OperatorGroup{
+				Spec: operatorsv1.OperatorGroupSpec{
+					StaticProvidedAPIs: true,
+				},
+			},
+			required: operatorsv1.OperatorGroup{
+				Spec: operatorsv1.OperatorGroupSpec{
+					StaticProvidedAPIs: false,
+				},
+			},
+
+			expectedModified: true,
+			expected: operatorsv1.OperatorGroup{
+				Spec: operatorsv1.OperatorGroupSpec{
+					StaticProvidedAPIs: false,
+				},
+			},
+		},
+		{
+			name: "implicit upgrade strategy",
+			existing: operatorsv1.OperatorGroup{
+				Spec: operatorsv1.OperatorGroupSpec{
+					UpgradeStrategy: operatorsv1.UpgradeStrategyUnsafeFailForward,
+				},
+			},
+			required: operatorsv1.OperatorGroup{
+				Spec: operatorsv1.OperatorGroupSpec{},
+			},
+
+			expectedModified: true,
+			expected: operatorsv1.OperatorGroup{
+				Spec: operatorsv1.OperatorGroupSpec{
+					UpgradeStrategy: operatorsv1.UpgradeStrategyDefault,
+				},
+			},
+		},
+		{
+			name: "same upgrade strategy",
+			existing: operatorsv1.OperatorGroup{
+				Spec: operatorsv1.OperatorGroupSpec{
+					UpgradeStrategy: operatorsv1.UpgradeStrategyUnsafeFailForward,
+				},
+			},
+			required: operatorsv1.OperatorGroup{
+				Spec: operatorsv1.OperatorGroupSpec{
+					UpgradeStrategy: operatorsv1.UpgradeStrategyUnsafeFailForward,
+				},
+			},
+
+			expectedModified: false,
+			expected: operatorsv1.OperatorGroup{
+				Spec: operatorsv1.OperatorGroupSpec{
+					UpgradeStrategy: operatorsv1.UpgradeStrategyUnsafeFailForward,
+				},
+			},
+		},
+		{
+			name: "different upgrade strategy",
+			existing: operatorsv1.OperatorGroup{
+				Spec: operatorsv1.OperatorGroupSpec{
+					UpgradeStrategy: operatorsv1.UpgradeStrategyUnsafeFailForward,
+				},
+			},
+			required: operatorsv1.OperatorGroup{
+				Spec: operatorsv1.OperatorGroupSpec{
+					UpgradeStrategy: operatorsv1.UpgradeStrategyDefault,
+				},
+			},
+
+			expectedModified: true,
+			expected: operatorsv1.OperatorGroup{
+				Spec: operatorsv1.OperatorGroupSpec{
+					UpgradeStrategy: operatorsv1.UpgradeStrategyDefault,
+				},
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			defaultOperatorGroup(&test.existing, test.existing)
+			defaultOperatorGroup(&test.expected, test.expected)
+			modified := pointer.BoolPtr(false)
```

Comment:
```suggestion
			modified := pointer.Bool(false)
```

Reply:
Fixed in the new commit https://github.com/openshift/cluster-version-operator/commit/2f0078a35636946231d5153645061744489cb3fc.